libs/ui/src/ui_strings/read_on_load.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { useAudioContext } from './audio_context';
+
+export interface ReadOnLoadProps {
+  children: React.ReactNode;
+  className?: string;
+}
+
+/**
+ * Returns the react-router location state if a react-router context exists,
+ * without failing if there isn't one available.
+ *
+ * Allows us to conditionally use router-aware logic in {@link ReadOnLoad}
+ * without requiring a react-router context.
+ */
+function useLocationIfAvailable() {
+  try {
+    return useLocation();
+  } catch {
+    return undefined;
+  }
+}
+
+/**
+ * On initial render, this triggers an audio read-out of any descendant
+ * `UiString` elements rendered within (in order of appearance in the DOM), if
+ * audio playback is enabled.
+ *
+ * Re-triggers a read-out on subsequent route changes, if applicable, or whenever
+ * audio playback is newly enabled.
+ *
+ * NOTE: Intended for use as a single instance on any given app screen. If
+ * multiple instances are present on screen, audio will be played only for
+ * content within the last `ReadOnLoad` instance.
```

Comment:
Would you classify multiple `ReadOnLoad` in a single render tree as a bug? If so, would it make sense to log/warn/throw?

Reply:
I don't think it's worth blocking. Go for it.

Reply:
SG! Will follow up once everything's wired up. Been thinking about the flip side of this, WRT people forgetting to add a `ReadOnLoad` instance to a voter-facing screen and how we might surface those as warnings -- can try to address both issues in the same task